js/main.js

```diff
@@ -1,45 +1,491 @@
+const PLAYER_BOARD_SECTION = document.querySelector(".player-board-section");
+const CPU_BOARD_SECTION = document.querySelector(".cpu-board-section");
+const activePlayerEl = document.querySelector("h2");
+const verticalCheckboxEl = document.querySelector("input");
+const verticalLabelEl = document.querySelector("label");
 /*----- event listeners -----*/
+CPU_BOARD_SECTION.addEventListener("click", handleMove);
+PLAYER_BOARD_SECTION.addEventListener("click", playerMouseClick);
+playAgainBtn.addEventListener("click", resetGame);
+verticalCheckboxEl.addEventListener("change", (e) => {
+  if (e.target.checked) {
+    verticalCheckbox = true;
+  } else {
+    verticalCheckbox = false;
+  }
+});
 /*----- functions -----*/
 render();
 
+//Initial rendering of the divs and header.
 function render() {
   createBoard("player");
   createBoard("cpu");
+  renderShips(CPU_BOARD);
+  activeTurn();
+}
+
+//Resetting the game to its base state after the match has ended
+function resetGame() {
+  playerShips = JSON.parse(JSON.stringify(BASE_PLAYER_SHIP_STATES));
+  cpuShips = JSON.parse(JSON.stringify(BASE_CPU_SHIP_STATES));
+  turn = 1;
+  winner = null;
+  LOCATIONS.clear();
+  resetBoard(PLAYER_BOARD, PLAYER_DIVS);
+  resetBoard(CPU_BOARD, CPU_DIVS);
+  if (document.body.contains(playAgainBtn))
+    document.querySelector("body").removeChild(playAgainBtn);
+  render();
 }
 
+//Creating the game board
 function createBoard(str) {
-  boardDivs.forEach((indArr, outerIdx) => {
+  PLAYER_BOARD.forEach((indArr, outerIdx) => {
     indArr.forEach((arrEl, innerIdx) => {
       const div = document.createElement("DIV");
-      div.className = "board";
-      if (innerIdx === 0) {
-        div.innerHTML = 'c' + (outerIdx+1);
-      } else
-        div.innerHTML = `c${outerIdx + 1}r${String.fromCharCode(
-          "a".charCodeAt(0) + innerIdx - 1
-        )}`;
+      div.innerHTML = "";
+      //Used to start at row A and increment to the next character in the alphabet
+      div.className = `board ${String.fromCharCode(
+        "a".charCodeAt(0) + innerIdx
+      )}${outerIdx + 1}`;
+      //Placing an active game space into an array to be accessed
+      if (parseInt(div.innerHTML) !== outerIdx + 1) {
+        str === "player" ? PLAYER_DIVS.push(div) : CPU_DIVS.push(div);
+      }
+      // Checking if the createBoard function was called for the player or the board to append the divs to the correct grid
       str === "player"
-        ? playerBoardSection.appendChild(div)
-        : cpuBoardSection.appendChild(div);
+        ? PLAYER_BOARD_SECTION.prepend(div)
+        : CPU_BOARD_SECTION.appendChild(div);
     });
   });
 }
+
+//Handling the player clicks for placing ships
+function playerMouseClick(evt) {
+  if (evt.target.tagName === "INPUT" || evt.target.tagName === "LABEL" || shipCounter > 6) return;
+  let clickedDiv = PLAYER_DIVS.indexOf(evt.target);
+  let rows = parseInt(clickedDiv.toString().charAt(0));
```

Comment:
This logic to get row/column indices from the `divIdx` is used a couple times throughout here. You should probably move this logic to a functional that returns row/col as an array or object or something. `get2DPosition` or something maybe?